src/Private/Console.ps1

```diff
@@ -0,0 +1,987 @@
+            '━' * $PodeContext.Server.Console.DividerLength  # Repeat the '━' character
+        }
+        else {
+            '-' * $PodeContext.Server.Console.DividerLength # Repeat the '-' as a fallback
+        }
+
+        # Write the divider with the chosen style
+        Write-PodeHost $dividerChar -ForegroundColor $dividerColor -Force:$Force
+    }
+    else {
+        Write-PodeHost
+    }
+}
+
+
+
+<#
+.SYNOPSIS
+    Displays information about the endpoints the Pode server is listening on.
+
+.DESCRIPTION
+    The `Show-PodeConsoleEndpointsInfo` function checks the Pode server's `EndpointsInfo`
+    and displays details about each endpoint, including its URL and any specific flags
+    such as `DualMode`. It provides a summary of the total number of endpoints and the
+    number of general threads handling them.
+
+.PARAMETER Force
+    Overrides the -Quiet flag of the server.
+
+.EXAMPLE
+    Show-PodeConsoleEndpointsInfo
+
+    This command will output details of all endpoints the Pode server is currently
+    listening on, including their URLs and any relevant flags.
+
+.NOTES
+    This function uses `Write-PodeHost` to display messages, with the `Yellow` foreground
+    color for the summary and other appropriate colors for URLs and flags.
+#>
+function Show-PodeConsoleEndpointsInfo {
+    param(
+        [switch]
+        $Force
+    )
+
+    # Default colors if not set
+    if ($null -ne $PodeContext.Server.Console.Colors.EndpointsHeader) {
+        $headerColor = $PodeContext.Server.Console.Colors.EndpointsHeader
+    }
+    else {
+        $headerColor = [System.ConsoleColor]::Yellow
+    }
+
+    if ($null -ne $PodeContext.Server.Console.Colors.Endpoints) {
+        $endpointsColor = $PodeContext.Server.Console.Colors.Endpoints
+    }
+    else {
+        $endpointsColor = [System.ConsoleColor]::Cyan
+    }
+
+    # Return early if no endpoints are available
+    if ($PodeContext.Server.EndpointsInfo.Length -eq 0) {
+        return
+    }
+
+    # Display header
+    Write-PodeHost ($PodeLocale.listeningOnEndpointsMessage -f $PodeContext.Server.EndpointsInfo.Length, $PodeContext.Threads.General) -ForegroundColor $headerColor -Force:$Force
+
+    # Write a horizontal divider line to the console.
+    Write-PodeHostDivider -Force $true
+    $disabled = ! (Test-PodeServerIsEnabled)
+    # Display each endpoint with extracted protocol
+    $PodeContext.Server.EndpointsInfo | ForEach-Object {
+        # Extract protocol from the URL
+        $protocol = ($_.Url -split ':')[0].ToUpper()
+
+        # Determine protocol label
+        $protocolLabel = switch ($protocol) {
+            'HTTP' { 'HTTP      ' }
+            'HTTPS' { 'HTTPS     ' }
+            'WS' { 'WebSocket ' }
+            'SMTP' { 'SMTP      ' }
+            'SMTPS' { 'SMTPS      ' }
+            'TCP' { 'TCP       ' }
+            'TCPS' { 'TCPS      ' }
+            default { 'UNKNOWN   ' }
+        }
```

Comment:
Is this necessary, as every endpoint printed is already prepended with the protocol? `https://` etc.

Reply:
It’s looking good 😀